src/sonic-yang-models/yang-models/sonic-vrrp.yang

```diff
@@ -0,0 +1,283 @@
+                            path "/spc:sonic-portchannel-interface/spc:PORTCHANNEL_INTERFACE/spc:PORTCHANNEL_INTERFACE_LIST/spc:name";
+                        }
+                        type leafref {
+                            path "/svlan:sonic-vlan/svlan:VLAN_INTERFACE/svlan:VLAN_INTERFACE_LIST/svlan:name";
+                        }
+                        type leafref {
+                            path "/svlansubint:sonic-vlan-sub-interface/svlansubint:VLAN_SUB_INTERFACE/svlansubint:VLAN_SUB_INTERFACE_LIST/svlansubint:name";
+                        }
+                    }
+                }
+
+                leaf idkey {
+                    type uint8 {
+                    range "1..255";
+                    }
+                }
+
+                leaf vrid {
+                    type uint8 {
+                    range "1..255";
+                    }
+                }
+
+                leaf version {
+                    type string {
+                        pattern '2|3';
+                    }
+                }
+
+                leaf pre_empt {
+                    type string {
+                        pattern 'True|False';
+                    }
+                }
+
+                leaf use_v2_checksum {
+                    type string {
+                        pattern 'True|False';
+                    }
+                }
+
+                leaf priority {
+                    type uint8 {
+                        range "1..254";
+                    }
+                    default 100;
+                }
+
+                leaf adv_interval {
+                    type uint8 {
+                        range "1..254";
+                    }
+                    default 1;
+                }
+
+                leaf-list vip {
+                    max-elements 4;
+                    type inet:ipv4-address;
+                }
+            }
+        }
+
+        container VRRP_TRACK {
+            description
+            "Configures VRRP track list.";
+
+            list VRRP_TRACK_LIST {
+
+                key "baseifname idkey trackifname";
+
+                leaf baseifname {
+                    type leafref {
+                        path "../../../VRRP/VRRP_LIST/ifname";
+                    }
+                }
+
+                leaf idkey {
+                    type leafref {
+                        path "../../../VRRP/VRRP_LIST/idkey";
+                    }
+                }
+
+                leaf trackifname {
+                    type union {
+                        type leafref {
+                            path "/sprt:sonic-port/sprt:PORT/sprt:PORT_LIST/sprt:name";
+                        }
+                        type leafref {
+                           path "/spc:sonic-portchannel/spc:PORTCHANNEL/spc:PORTCHANNEL_LIST/spc:name";
+                        }
+                        type leafref {
+                            path "/svlan:sonic-vlan/svlan:VLAN/svlan:VLAN_LIST/svlan:name";
```

Comment:
this has to be comment out due to back link in yang issue.
please check sample:
https://github.com/sonic-net/sonic-buildimage/blob/master/src/sonic-yang-models/yang-models/sonic-vxlan.yang#L84-L86